src/main/java/no/nav/log/SOEPreventionFilter.java

```diff
@@ -0,0 +1,70 @@
+package no.nav.log;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ThrowableProxy;
+import ch.qos.logback.classic.turbo.TurboFilter;
+import ch.qos.logback.core.spi.FilterReply;
+import org.slf4j.Marker;
+
+import java.util.Arrays;
+
+import static no.nav.log.SOEPreventionFilter.ExceptionLoopDetector.hasLoop;
+
+/**
+ * Dette filteret er en hack for å unngå java.lang.StackOverflowError siden spring boot ikke støtter logback 1.3 P.T.
+ *
+ * ref:
+ * https://jira.qos.ch/browse/LOGBACK-1454
+ * https://github.com/spring-projects/spring-boot/issues/12649
+ *
+ * obs: Hele denne klassen kan fjernes når vi er over på en versjon av logback som takler exceptions med sirkusreferanser
+ */
+public class SOEPreventionFilter extends TurboFilter {
+
+    @Override
+    public FilterReply decide(
+            Marker marker,
+            Logger logger,
+            Level level,
+            String msg,
+            Object[] objects,
+            Throwable t
+    ) {
+        if (t != null) { // log.level("", ex)
+            if (hasLoop(t)) {
+                logger.log(marker, logger.getName(), Level.toLocationAwareLoggerInteger(level), msg, objects, new RuntimeException(t.getMessage()));
+                return FilterReply.DENY;
+            }
+        } else { // log.level("lorum {} ipsum {}", str1, obj2, ex)
+            Object lastArg = objects == null ? null : objects[objects.length - 1];
+            if (lastArg instanceof Throwable) {
+                if (hasLoop((Throwable) lastArg)) {
+                    Object[] copy = Arrays.copyOf(objects, objects.length);
+                    copy[objects.length - 1] = new RuntimeException(((Throwable) lastArg).getMessage());
+                    logger.log(marker, logger.getName(), Level.toLocationAwareLoggerInteger(level), msg, copy, t);
+                    return FilterReply.DENY;
+                }
+
+                return FilterReply.NEUTRAL;
+            }
+        }
+
+        return FilterReply.NEUTRAL;
+    }
+
+    static class ExceptionLoopDetector {
+
+        /**
+         * test om throwable lar seg proxye i logback
+         */
+        static boolean hasLoop(Throwable t) {
```

Comment:
brukte lang tid i dag på en altfor komplisert måte å analysere t etter sirkulære refs. Fant til slutt ut at det var like greit å bare gjøre en test opprettelse av ThrowableProxy

Reply:
Har du prøvd https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Throwables.java#L297

Reply:
For å detektere loops

Reply:
Føles veldig feil å forvente stackoveflow, men hvis det er løsningen så blir det det. Veldig god ide med TurboFilter!

Reply:
prøvde litt forskjellig, men man må sjekke suppressed i tillegg, så det ble en del kode. Dette virker tryggere og enklere. 
Kan evt utvide catchen hvis vi er redd for at det brekker på andre måter.

Reply:
enig at det er litt shady å forvente stackoverflowerror. God ide med å prøve Throwables. Jeg forsøkte et par andre ExceptionUtils, men var ikke klar over Throwables. Mener å huske at spring internt har en klasse som detekterer loops, men klarte ikke finne den igjen. Jeg gjør et forsøk med Throwables nå på morran :)

Reply:
jorde et forsøk til men måtte gi opp. Virker som det er sirkulært fra t -> cause -> supressed[0] -> t.
Det gjør det litt komplisert å lete. Siden vi har fått en throwable som arg inn her så vet vi at det kommer til å bli opprettet en ThrowableProxy, så da tenker jeg det er greit å prøve å kalle constructoren selv først. Utvidet catch til å fange mer enn bare SOE i tilfelle.